ReactiveCocoaTests/Swift/SignalLifetimeSpec.swift

```diff
@@ -210,6 +210,91 @@ class SignalLifetimeSpec: QuickSpec {
 				expect(signal).to(beNil())
 			}
 		}
+
+		describe("observe") {
+			/// Used to detect the lifetime of `Signal` handlers.
+			final class Token {
+				private let deallocHandler: () -> ()
+
+				init(deallocHandler: () -> ()) {
+					self.deallocHandler = deallocHandler
+				}
+
+				func test() {}
+
+				deinit { self.deallocHandler() }
+			}
+
+
+			var signal: Signal<Int, TestError>!
+			var observer: Signal<Int, TestError>.Observer!
+
+			var tokenWasDeallocated: Bool!
+
+			var token: Token? = nil
+			weak var weakToken: Token?
+
+			func expectTokenNotDeallocated() {
+				expect(tokenWasDeallocated) == false
+				expect(weakToken).toNot(beNil())
```

Comment:
These two tests seem redundant. Shouldn't it be enough to just test a weak reference? In which case, we could remove `Token` and use an object from Foundation or Swift that supports weak references?